crates/pipeline-manager/src/runner/pipeline_automata.rs

```diff
@@ -1287,13 +1230,57 @@ impl<T: PipelineExecutor> PipelineAutomaton<T> {
                     State::TransitionToRunning
                 }
             }
-            StatusCheckResult::Unavailable => {
-                if pipeline.deployment_status == PipelineStatus::Unavailable {
-                    State::Unchanged
-                } else {
-                    State::TransitionToUnavailable
+            StatusCheckResult::Initializing => match pipeline.deployment_status {
+                // Pipeline is Initializing but INITIALIZING_TIMEOUT has expired
+                PipelineStatus::Initializing
+                    if Self::has_timeout_expired(
```

Comment:
I think the timeout behavior for Initializing doesn't have to be preserved, the pipeline can just return an error if during initialization it won't be able to finish. What do you think @ryzhyk ?

Reply:
yes, we want to drop this timeout, so the pipeline can be in this INITIALIZING state indefinitely.

Reply:
this is done in #4367 , will update it here once that is merged.

Reply:
done it here.